File: pgscala-converters/src/main/scala/hr/element/pgscala/converters/PGNullableDateConverter.scala
package hr.element.pgscala.converters

import org.joda.time.LocalDate

object PGNullableDateConverter extends PGTypeConverter[Option[LocalDate]] {
  def toPGString(value: Option[LocalDate]): String =
    value match {
      case Some(lD) =>
        PGDateConverter.toPGString(lD)
      case _ =>
        null
    }

  def fromPGString(value: String): Option[LocalDate] =
    if(value != null && value.nonEmpty) {
      Some(PGDateConverter.fromPGString(value))
    }
    else {
      None
    }

  override val PGType = java.sql.Types.DATE
}

File: pgjava-converters/src/main/java/hr/element/pgscala/converters/PGNullableDateConverter.java
package hr.element.pgscala.converters;

import org.joda.convert.*;

import org.joda.time.*;

public class PGNullableDateConverter {
  public static final String pgType = "date";

  @ToString
  public static String toPGString(final LocalDate lD) {
    return null == lD ? null : lD.toString();
  }

  @FromString
  public static LocalDate fromPGString(final String dT) {
    return null == dT ? null : new LocalDate(dT);
  }
}
